eth/gasprice/feehistory.go

```diff
@@ -88,10 +88,10 @@ func (oracle *Oracle) processBlock(bf *blockFees, percentiles []float64) {
 	if bf.results.baseFee = bf.header.BaseFee; bf.results.baseFee == nil {
 		bf.results.baseFee = new(big.Int)
 	}
-	if chainconfig.IsLondon(big.NewInt(int64(bf.blockNumber + 1))) {
+	if chainconfig.IsLondon(big.NewInt(int64(bf.blockNumber+1))) && chainconfig.EnableEIP2718 && chainconfig.EnableEIP1559 {
 		bf.results.nextBaseFee = misc.CalcBaseFee(chainconfig, bf.header)
```

Comment:
Won't `misc.CalcBaseFee` return `nil` anyway?

Reply:
Yes, it return `nil`, there are a some other places with similar logic, I think better add this if too, so it will be easier to understand in future for people who not familiar with this

Reply:
then what about 
+ core/tx_pool.go
+ eth/catalyst/api.go